src/main/java/com/stuypulse/robot/subsystems/Edwin.java

```diff
@@ -0,0 +1,229 @@
+import static com.stuypulse.robot.Constants.EdwinSettings.*;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class Edwin extends Robot {
+    
+    private final MotorControllerGroup left;
+    private final MotorControllerGroup right;
+    
+    private final DifferentialDrive drivetrain;
+
+    private final Encoder leftGreyhill;
+    private final Encoder rightGreyhill;
+
+    private final Controller leftController, rightController;
+    private double leftTargetSpeed, rightTargetSpeed;
+
+    private final DifferentialDriveOdometry odometry;
+    private final DifferentialDriveKinematics kinematics;
+    private final AHRS navx;
+
+    private final Field2d field;
+
+    public Edwin() {
+        left = new MotorControllerGroup(
+            new CANSparkMax(Ports.Edwin.LEFT_TOP, MotorType.kBrushless), 
+            new CANSparkMax(Ports.Edwin.LEFT_MIDDLE, MotorType.kBrushless),
+            new CANSparkMax(Ports.Edwin.LEFT_BOTTOM, MotorType.kBrushless)
+        );
+
+        right = new MotorControllerGroup(
+            new CANSparkMax(Ports.Edwin.RIGHT_TOP, MotorType.kBrushless), 
+            new CANSparkMax(Ports.Edwin.RIGHT_MIDDLE, MotorType.kBrushless),
+            new CANSparkMax(Ports.Edwin.RIGHT_BOTTOM, MotorType.kBrushless)
+        );
+
+        drivetrain = new DifferentialDrive(left, right);
+
+        leftGreyhill = new Encoder(Ports.Edwin.Encoders.LEFT_A, Ports.Edwin.Encoders.LEFT_B);
+        rightGreyhill = new Encoder(Ports.Edwin.Encoders.RIGHT_A, Ports.Edwin.Encoders.RIGHT_B);
+        
+        setGreyhillDistancePerPulse(EdwinSettings.Encoders.GREYHILL_DISTANCE_PER_PULSE);
+
+        leftController = new Feedforward.Drivetrain(Motion.Feedforward.kS, Motion.Feedforward.kV, Motion.Feedforward.kA).velocity()
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+        
+        rightController = new Feedforward.Drivetrain(Motion.Feedforward.kS, Motion.Feedforward.kV, Motion.Feedforward.kA).velocity()
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+
+        leftTargetSpeed = 0;
+        rightTargetSpeed = 0;
+
+        odometry = new DifferentialDriveOdometry(getRotation2d());
+        kinematics = new DifferentialDriveKinematics(EdwinSettings.TRACK_WIDTH);
+        navx = new AHRS(SPI.Port.kMXP);
+
+        right.setInverted(true);
+        reset(EdwinSettings.Odometry.STARTING_POSITION);
```

Comment:
make this just new POse2d()... all the path following for this project is relative